db/migrate/20160706110723_supersede_previous_unpublished_items.rb

```diff
@@ -0,0 +1,25 @@
+class SupersedePreviousUnpublishedItems < ActiveRecord::Migration
+  def up
+    state_histories = State
+      .where(name: %w(published unpublished))
+      .joins(:content_item)
+      .joins("INNER JOIN translations t on t.content_item_id = content_items.id")
+      .joins("INNER JOIN user_facing_versions u on u.content_item_id = content_items.id")
+      .group(:content_id, :locale)
+      .having("count(content_id) > 1")
+      .pluck("json_agg((states.id, u.number))")
+      .map do |results|
+        results.map { |row| { state_id: row["f1"], version: row["f2"] } }
+      end
+
+    states_to_supersede = state_histories.map do |history|
```

Comment:
I think this should be `flat_map`? This will produce an array of arrays. The outer array represents content_ids, the inner array is the content_item_ids. `flat_map` will just give you a single flat array of content_ids. This probably doesn't matter in practice except your `.count` below will be more accurate.